l10n_br_account/models/account_move_line.py

```diff
@@ -216,13 +216,51 @@ def create(self, vals_list):
                     )
                 )
         self._inject_shadowed_fields(vals_list)
+
+        # This reordering is crucial to ensure accurate linkage between
+        # account.move.line (aml) and the fiscal document line. In the fiscal create a
+        # fiscal document line, leaving only those that should be created. Proper
+        # ordering is essential as mismatches between the order of amls and the
+        # manipulated vals_list of fiscal documents can lead to incorrect linkages.
+        # For example, if vals_list[0] in amls does not match vals_list[0] in the
+        # fiscal document (which is a manipulated vals_list), it results in erroneous
+        # associations.
+        if len(vals_list) > 1 and self.env.company.country_id.code == "BR":
+            # Store the original order using a copy of the data
+            original_vals_list = list(vals_list)
+            # Reorder vals_list for processing based on the presence of
+            # 'fiscal_operation_line_id'
+            vals_list = sorted(
+                vals_list,
+                key=lambda x: x.get("fiscal_operation_line_id") is not False,
+                reverse=True,
```

Comment:
não é melhor fazer o sort com False (em vez de not False) e remover esse reverse?

Reply:
testando local

Reply:
feito

Reply:
eu tb ainda nao olhei a fundo, mas essa reordenacao no fim parece bem laboriosa e esse lance de refazer um browse é meio feio... Se não tiver opção melhor talvez teremos que fazer e ainda assim talvez sera melhor do que o unlink de antes, mas  eu suspeito que talvez é possível melhorar essa re-ordenacão.

Reply:
cara .. o unlink não parece uma boa não.. acho que podemos melhorar aqui mas voltar para o unlink nao parece uma boa opção nao.

Reply:
> eu tb ainda nao olhei a fundo, mas essa reordenacao no fim parece bem laboriosa e esse lance de refazer um browse é meio feio... Se não tiver opção melhor talvez teremos que fazer e ainda assim talvez sera melhor do que o unlink de antes, mas eu suspeito que talvez é possível melhorar essa re-ordenacão.

tbm achei feio esse browse.. mas foi oq consegui por ora :D heheheh vou passar mais uma vez pelo gpt apesar que ele anda meio burro as vezes :D

Reply:
voltando sobre o unlink, eu só olhei o impacto nos campos ind_final e ind_pres.. mas parece que tem impacto em vários outros.

Reply:
ao invés do browse.. uma alternativa poderia ser algo assim

results = f"account.move.line({', '.join(map(str, all_ids))})"

Reply:
nao fica muito legal tbm assim nao :D